### PR TITLE
JBPM-5389: Some cleans for Shapes and Views.

### PR DESCRIPTION
**Pull request by romartin**
Hey @manstis , @hasys 

This is a quite big clean of abstract classes and implementations for the stunner Shapes, was quite impossilble to understand, either for me that wrote it :/ Well just a quick clean I think it worth and it's better for all of us. Added some new unit tests as well.

Once this commit gets merged I will provide a second PR for this ticket, that includes the new SVG support stuff. So I have tried to split the work as much as poosible this time ;)

You will notice that for the BPMN side, some of the shape's icons are different, as I have removed the static/dynamic icon shapes (which  tbh have no much sense), anyway will be replaced on the second commit by svg file icons.

Thanks!
Roger

### CODE REVIEW

**Comment by romartin** [comment]
jenkins retest this

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/ShapeViewHandler.java [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+public class ShapeViewHandler<V extends ShapeView> {
+
+    private final V view;
+
+    public ShapeViewHandler(final V view) {
+        this.view = view;
+    }
+
+    public void applyTitle(final String title,
+                           final MutationContext mutationContext) {
```
Comment (hasys) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/ShapeViewHandler.java]: mutationContext parameter is never used in this Class (but used in 8 method declaration). I suppose add this parameter in the future will be easier than clean up it.

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/NodeShapeImpl.java [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+            this.state = shapeState;
+            if (ShapeState.SELECTED.equals(shapeState)) {
+                applySelectedState();
+            } else if (ShapeState.HIGHLIGHT.equals(shapeState)) {
+                applyHighlightState();
+            } else if (ShapeState.INVALID.equals(shapeState)) {
+                applyInvalidState();
+            } else {
+                applyNoneState(_strokeColor,
+                               null != this._strokeWidth ? this._strokeWidth : 1,
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/NodeShapeImpl.java]: yeah good point thx! refactored

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/ShapeDefViewHandler.java [hunk 3]
```diff
@@ -0,0 +1,159 @@
...
+        // Apply border alpha.
+        final double _strokeAlpha = getBorderAlpha(element);
+        viewHandler.applyBorderAlpha(_strokeAlpha,
+                                     mutationContext);
+    }
+
+    public void applyTitle(final String title,
+                           final W element,
+                           final MutationContext mutationContext) {
+        if (title.trim().length() > 0) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/ShapeDefViewHandler.java]: well I've removed this condition, my thoughts are: let's keep it simple and let the name be empty, so delegate that as much as possible to each bean and how it should be displayed.

File: kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/main/java/org/kie/workbench/common/stunner/shapes/client/BasicConnectorShape.java [hunk 4]
```diff
@@ -0,0 +1,87 @@
...
+                               V view) {
+        super(shapeDef,
+              view);
+    }
+
+    @Override
+    public void applyState(ShapeState shapeState) {
+        if (!this.state.equals(shapeState)) {
+            this.state = shapeState;
+            if (ShapeState.SELECTED.equals(shapeState)) {
```
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-client/kie-wb-common-stunner-shapes/kie-wb-common-stunner-shapes-client/src/main/java/org/kie/workbench/common/stunner/shapes/client/BasicConnectorShape.java]: good point as well, refactored ;)

File: kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/ShapeViewHandler.java (hunk 1 above)
Comment (romartin) [inline kie-wb-common-stunner/kie-wb-common-stunner-core/kie-wb-common-stunner-commons/kie-wb-common-stunner-client-common/src/main/java/org/kie/workbench/common/stunner/core/client/shape/impl/ShapeViewHandler.java]: well in fact this mutation context instance was being used not so far ago when I was applying animations on shapes on different behaviors, for example, when an start event was being added into the canvas it was being using some bounce effects for its radius and things like that  ... but at that point there are already some missing points about the shapes stuff and the ways to interact with UX, so I just removed that animated support. 
I did not remove this mutationContext as I'm planning add animation support at some point. Not sure guys... WDYT?

**Comment by romartin** [comment]
@hasys @manstis thanks for the review guys, added a new commit with that fixes.

**Comment by manstis** [comment]
@hasys Could you please approve before merge? Thanks.

**Comment by hasys** [comment]
@romartin, @manstis, looks good now, :+1:

**Pull request merged**